map2loop/mapdata.py

```diff
@@ -689,161 +690,46 @@ def check_map(self, datatype: Datatype):
                 The datatype to check
         """
         func = None
+        #check and parse geology data
         if datatype == Datatype.GEOLOGY:
+            validity_check, message = check_geology_fields_validity(mapdata = self)
+            if validity_check:
+                logger.error(f"Datatype GEOLOGY - data validation failed: {message}")
+                raise ValueError(f"Datatype GEOLOGY - data validation failed: {message}")
             func = self.parse_geology_map
+            
+        #check and parse structure data
         elif datatype == Datatype.STRUCTURE:
+            validity_check, message = check_structure_fields_validity(mapdata = self)
+            if validity_check:
+                logger.error(f"Datatype STRUCTURE - data validation failed: {message}")
+                raise ValueError(f"Datatype STRUCTURE - data validation failed: {message}")
             func = self.parse_structure_map
+        
+        #check and parse fault data
         elif datatype == Datatype.FAULT:
+            validity_check, message = check_fault_fields_validity(mapdata = self)
+            if validity_check:
+                logger.error(f"Datatype FAULT - data validation failed: {message}")
```

Comment:
I'm not sure if there is too much duplication in the error messages. The validity checker prints out nice error messages. I think it would be enough to just raise the exception `Datatype FAULT - data validation failed`